python/lsst/the/monster/assemble.py

```diff
@@ -150,42 +204,183 @@ def run(self,
+
+            orig_flux = gaia_stars_all[flux_col]
+            orig_flux_err = gaia_stars_all[flux_col + 'Err']
+            slr_model_flux = colorterm_spline.apply(
+                gaia_stars_all[flux_col_1],
+                gaia_stars_all[flux_col_2],
+                orig_flux,
+            )
+            # Rescale flux error to keep S/N constant
+            slr_model_flux_err = slr_model_flux * (orig_flux_err/orig_flux)
+            # apply offsets to pin SLR to sdss_u_from_gaiaXPu
+            slr_model_flux, slr_model_flux_err = self.apply_u_band_offsets(
+                gaia_stars_all=gaia_stars_all,
+                slr_model_flux=slr_model_flux,
+                slr_model_flux_err=slr_model_flux_err
+            )
+
+            for target_system_name, band in target_systems_u_transform:
+                # we take our SDSS_u flux and transform to target system
+                colorterm_file_string = 'SDSS_to_'+str(target_system_name)+'_band'
+                colorterm_spline = self.get_colorterm_spline(colorterm_file_string, band)
+                flux_col = f"monster_{target_system_name}_{band}_flux"
+                # SDSS_to_SDSS_band_u source_color_field_1 = psfMag_g_flux
+                # should we change to just g? so we can use
+                # colorterm_spline.source_color_field_1 below
+                if colorterm_spline.source_color_field_1 == 'psfMag_g_flux':
+                    colorterm_spline.source_color_field_1 = 'g'
+                    colorterm_spline.source_color_field_2 = 'r'
+                flux_col_1 = f'monster_DES_{colorterm_spline.source_color_field_1}_flux'
+                flux_col_2 = f'monster_DES_{colorterm_spline.source_color_field_2}_flux'
+                if self.testing_mode:
+                    flux_col = flux_col.replace("monster_DES", "monster_TestDES")
+                    flux_col_1 = flux_col_1.replace("monster_DES", "monster_TestDES")
+                    flux_col_2 = flux_col_2.replace("monster_DES", "monster_TestDES")
+
+                orig_flux = slr_model_flux
+                orig_flux_err = slr_model_flux_err
+
+                model_flux = colorterm_spline.apply(
+                    gaia_stars_all[flux_col_1],
+                    gaia_stars_all[flux_col_2],
+                    orig_flux,
+                )
+                # Rescale flux error to keep S/N constant
+                model_flux_err = model_flux * (orig_flux_err/orig_flux)
+                model_flux_not_nan = np.isfinite(model_flux)
+                flag = model_flux_not_nan
+
+                gaia_stars_all[flux_col][flag] = model_flux[flag]
+                gaia_stars_all[flux_col + "Err"][flag] = model_flux_err[flag]
+                # Update the flags to denote which survey the flux came
+                gaia_stars_all[flux_col.replace('flux', 'source_flag')][flag] = FLAG_DICT["SLR"]
+
+        # next perform non SLR u band transformations to target systems
```

Comment:
It would be worth adding a comment about which catalogs are in play.  At the same time, I don't see where the SDSS u band is used (if available).

Reply:
added a comment on default u-band catalogs here and included sdssu as a default catalog